content/acra/security-controls/searchable-encryption/_index.md

```diff
@@ -30,8 +30,55 @@ SELECT ... FROM ... WHERE encrypted_column LIKE "prefix%"
 AcraServer supports `SELECT/UPDATE/INSERT/DELETE` SQL statements for searchable encryption queries.
 {{< /hint >}}
 
+AcraServer/AcraTranslator also provide the ability to join tables rows using `JOIN` queries through searchable functionality.
+
+For example, queries like that:
+```
+SELECT ... FROM ... join table on table.searchable_encrypted_column = "value"
+```
+
+Moreover, table rows could be joined not only via `ON searchable_encrypted_column = <value>`, but via the searchable column of another table as well.
+```
+SELECT ... FROM table1 join table on table1.searchable_encrypted_column = table2.searchable_encrypted_column
+```
+
+In such cases, it is important to have searchable data in both tables encrypted via the same ClientID to have the same calculated keyed hash, otherwise, rows won't be joined.
+
+Consider an example of modification of the searchable query received in AcraServer/AcraTranslator.
+
+Initial query received by AcraServer/AcraTranslator:
+```
+SELECT ... FROM ... WHERE searchable_column = "value"
+```
+
 Under the hood, AcraServer/AcraTranslator will calculate a keyed hash of plaintext data (`blind index`), then actually encrypt the data into AcraStruct/AcraBlock, then return `"blind index|AcraStruct or AcraBlock"` envelope to store in a database. Thus, the actual plaintext data is encrypted and searching is based on keyed hashes.
 
+The left expression of the WHERE statement is changed on `substring` operator over the searchable binary data column to extract and check matching with the calculated hash, represented as the right side expression.
+
+Ultimately, the final view of the searchable query constructed by AcraServer/AcraTranslator and sent to the database will look like this:
+```
+SELECT ... FROM ... WHERE substring(searchable_column, 1, <HMAC_size>) = X'HASH_value'`
```

Comment:
`X'<value>'` was true for MySQL (now it is 0x<hash_value>). As I remember, for postgresql Acra passes `'\x<hash_value>`.

Reply:
Yes, exactly, lets have several example queries for MySQL and PostgreSQL